src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] arrayList;
+    private int size = 0;
+    private int emptyPlaces = 10;
+    private final double ARRAY_ENLARGEMENT_PERCENTAGE = 0.5;
```

Comment:
The constant 'ARRAY_ENLARGEMENT_PERCENTAGE' should be static. Moreover, this value is a 'magic number' and should be a constant with an informative name. Also, the 'emptyPlaces' variable is not used consistently and should be removed in favor of a constant INITIAL_CAPACITY.